modules/sandboxed-containers-preparing-openshift-cluster.adoc

```diff
@@ -3,5 +3,102 @@
+* Each VM runs in a `qemu` process and hosts a `kata-agent` process. 
+
+Each VM is configured with a default amount of memory. Additional memory is hot-plugged into the VM for containers that explicitly request memory.
+
+* If a container runs without a given memory resource, it is able to consume free memory. It will do so until the total memory used by the VM reaches the default allocation. The guest and its I/O buffers also consume memory.
+* If a container is given a specific amount of memory, then that memory is hot-plugged into the VM before the container starts.
+* If a memory limit is specified, then the workload is terminated if it consumes more memory than the limit. If no memory limit is specified, the kernel  that is running on the virtual machine might run out of memory. If the kernel runs out of memory it might terminate other process on the virtual machine.
+
+.Indicative memory sizes
+
+The table below lists some typical values for resource allocation.
+
+[cols="2,2"]
+|===
+|Resource |Value
+
+|Memory allocated by default to a virtual machine | 2G
+|Guest Linux kernel memory usage at boot | ~110M
+|Memory used by the QEMU process (excluding VM memory) | ~30M
+|Memory used by the `virtiofsd` process (excluding VM I/O buffers) | ~10M
+|Memory used by the `containerd-shim-kata-v2` process | ~20M
+|File buffer cache data after running `dnf install` on Fedora | ~300M* ^[1]^
+|===
+[.small]
+--
+Total memory usage is properly accounted for by the memory utilization metrics, which only count that memory once.
+
+1. File buffers appear and are accounted for in multiple locations:
+* In the guest where it appears as file buffer cache.
+* In the `virtiofsd` daemon, mapped allow user-space file I/O operations.
+* In the QEMU process as guest memory.
+--
+
+link:https://kubernetes.io/docs/concepts/scheduling-eviction/pod-overhead/[Pod overhead] describes the amount of system resources that a pod on a node uses. You can get the current pod overhead for the `kata` runtime class by using `oc describe runtimeclass kata` as shown below.
+
+.Example
+[source,terminal]
+----
+$ oc describe runtimeclass kata
+Name:         kata
+[...]
+Metadata:
+[...]
+Overhead:
+  Pod Fixed:
+    Cpu:     250m
+    Memory:  350Mi
+[...]
+----
+
+You can change the pod overhead by changing the `spec.overhead` field for a `RuntimeClass`. For instance, if the configuration that you run for your containers consumes more than 350Mi of memory for the QEMU process and guest kernel data, you may alter the `RuntimeClass` overhead to suit your needs.
+
+[NOTE]
+====
+The specified default overhead values are supported by Red Hat. Changing default overhead values may result in technical issues that is not supported by Red Hat.
+====
+
+.Example
+[source,yaml]
+----
+kind: RuntimeClass
+apiVersion: node.k8s.io/v1
+metadata:
+    name: kata
+overhead:
+    podFixed:
+        memory: "500Mi"
+        cpu: "500m"
+----
+
+* The default allocation for virtual machines is 2Gi.
```

Comment:
These bullet points can be removed now that the table above provides the information.

Reply:
Are the values in the table the defaults? The text just calls them _typical values_.

Reply:
@mburke5678 The value of memory allocated to the VM is specified in a configuration file, so it is a default value. The other values are observed values on small workloads like `ngingx`. They vary by workload.

Reply:
@sagidlow I believe that the entire text could be dropped. It adds nothing compared to the table, and simply feels like a repeat of (somewhat secondary) information.

Reply:
@sagidlow To be precise, what I would drop starts with `The default allocation for virtual machines…` and ends with `when reporting the memory utilization metrics`